src/console/src/databank/tasks.py

```diff
@@ -29,7 +31,22 @@ def set_state(self, replication_value):
 
     def get_state(self):
         try:
-            return self.etl_config.replication_state["replication_value"]
+            state = self.etl_config.replication_state or {}
+
+            value = state.get("replication_value")
+            if not value:
+                return None
+
+            backfill = state.get("backfill", 0)
+
+            ts = value if isinstance(value, datetime) else parse_datetime(value)
+            if ts is None:
+                return value
+
+            ts -= timedelta(seconds=backfill)
+
+            return ts.strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
don't do this. do str(ts). Check this: if backfill is present, just assume value is in datetime format.